client/lib/sites-list/list.js

```diff
@@ -394,7 +394,7 @@ SitesList.prototype.setRecentlySelectedSite = function( siteID ) {
 };
 
 SitesList.prototype.getRecentlySelected = function() {
-	if ( ! this.recentlySelected.length ) {
+	if ( ! this.recentlySelected.length || ! this.initialized ) {
```

Comment:
Yes, it's not the cleanest stuff, we really need to tackle that sites-list refactor :)